packages/guides-restructured-text/src/RestructuredText/Directives/ConfigurationBlockDirective.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\RestructuredText\Directives;
+
+use phpDocumentor\Guides\Nodes\CodeNode;
+use phpDocumentor\Guides\Nodes\CollectionNode;
+use phpDocumentor\Guides\Nodes\Configuration\ConfigurationBlockNode;
+use phpDocumentor\Guides\Nodes\Configuration\ConfigurationTab;
+use phpDocumentor\Guides\Nodes\Node;
+use phpDocumentor\Guides\RestructuredText\Parser\BlockContext;
+use phpDocumentor\Guides\RestructuredText\Parser\Directive;
+use phpDocumentor\Guides\RestructuredText\Parser\Productions\Rule;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\String\Slugger\AsciiSlugger;
+use Symfony\Component\String\Slugger\SluggerInterface;
+
+use function assert;
+use function get_debug_type;
+use function str_replace;
+use function ucfirst;
+
+final class ConfigurationBlockDirective extends SubDirective
+{
+    /** @var array<string, string> */
+    private array $languageLabels = [];
+
+    private SluggerInterface $slugger;
+
+    /**
+     * @param Rule<CollectionNode> $startingRule
+     * @param array<array{language: string, label: string}> $languageLabels
+     */
+    public function __construct(
+        private LoggerInterface $logger,
+        Rule $startingRule,
+        array $languageLabels = [],
+    ) {
+        parent::__construct($startingRule);
+
+        $this->slugger = new AsciiSlugger();
+
+        foreach ($languageLabels as $item) {
+            $this->languageLabels[$item['language']] = $item['label'];
```

Comment:
:thinking: this doesn't look right. A single language, like `php`, should be able to be associated with several labels, like `attribute` or `annotation`.

Reply:
No, this is correct. We have 2 "maps":

* Alias to highlighting language
* Alias to label

E.g. `.. code-block:: php-attributes` is aliased to `php` in the highlighter, but has the `Attributes` label assigned to it. Maybe we can even combine all of this in one config option (if we move this directive to guides-code), e.g. `<language name="php-attributes" highlight-as="php" label="Attributes"/>`

Reply:
Oh I see, ok!